std/range/package.d

```diff
@@ -1404,7 +1418,10 @@ if (Ranges.length > 0 &&
                         assert(0, "Internal library error. Please report it.");
                     }
 
-                    return result;
+                    static if (bidirectional)
```

Comment:
How is this related?

Reply:
Further down, we used to mutate `result.backIndex` in the `static if (bidirectional)` branch. As `result` no longer exists, we have to provide a variable for that branch to mutate.